src/utils.jl

```diff
@@ -127,3 +127,22 @@ end
     end
     return
 end
+
+"""
+    with(row; patch...)
+    with(row, patches...)
+
+Return a `NamedTuple` by merging `row` (a `AbstractRow`-compliant value) with `patches` (one or more
+`AbstractRow`-compliant values) via `Base.merge`. In other words, this function is similar to
+`Base.merge(::NamedTuple, ::NamedTuple...)`, but accepts `AbstractRow`-compliant values intead of
+`NamedTuple`s.
+
+A convenience method `with(row; patch...) = with(row, patch)` is defined that enables `patch`
+fields to be specified as keyword arguments.
+"""
+with(row; patch...) = with(row, patch)
```

Comment:
```suggestion
with(row; patch...) = with(row, patch.data)
```
this will be more efficient as it skips wrapping a NamedTuple in a Row and then constructing a NamedTuple from that